tests/test_glue.py

```diff
@@ -108,3 +108,50 @@ def test_dev_endpoints_delete(self):
         client = session_factory().client("glue")
         dev_endpoints = client.get_dev_endpoints()["DevEndpoints"]
         self.assertFalse(dev_endpoints)
+
+
+class TestGlueTag(BaseTest):
+    def test_glue_tags(self):
+        session_factory = self.replay_flight_data("test_glue_tags")
+        policy = {
+            'name': 'glue-dev-endpoint',
+            'resource': 'glue-dev-endpoint',
+            'actions': [
+                {
+                    'type': 'tag',
+                    'key': 'abcd',
+                    'value': 'xyz'
+                },
+            ]
+        }
+        p = self.load_policy(
+            policy,
+            session_factory=session_factory)
+
+        resources = p.run()
+        client = session_factory().client("glue")
+        arn = p.resource_manager.generate_arn(resources[0]['EndpointName'])
+        tags = client.get_tags(ResourceArn=arn)
+
+        self.assertEqual(len(resources), 1)
```

Comment:
this should also assert that self.assertEqual(resources['Tags'], []) ie that a change was made by the action.

Reply:
Hi Kapil, in the assert after this, I am doing the check to see that the tag exists after tag test policy runs and after the untag test policy test I am checking that the tags are now empty.  ``self.assertEqual(tags.get('Tags'), {'abcd': 'xyz'})`` for tag test and then ``self.assertEqual(tags.get('Tags'), {})`` after untag test policy runs. Is that what you meant?

Reply:
The problem is that it’s verifying expected state after the policy executed without verifying initial state, so if initial state matched end state, the policy is effectively a no op. The resources returned by the policy are effectively pre action execution and can serve as a means of verifying initial conditions

Reply:
The problem is that it’s verifying expected state after the policy executed without verifying initial state, so if initial state matched end state, the policy is effectively a no op. The resources returned by the policy are effectively pre action execution and can serve as a means of verifying initial conditions

Reply:
ah ok I got what you mean now. But I don't think the resource returned by the policy currently returns tags. Ill add the logic in the GlueDevEndpoint class to get the tags.